pkg/auth/providers/common/ldap/ldap_util.go

```diff
@@ -39,35 +39,36 @@ func NewLDAPConn(servers []string, TLS, startTLS bool, port int64, connectionTim
 	var err error
 	var tlsConfig *tls.Config
 	ldapv3.DefaultTimeout = time.Duration(connectionTimeout) * time.Millisecond
-	// TODO implment multi-server support
-	if len(servers) != 1 {
-		return nil, errors.New("invalid server config. only exactly 1 server is currently supported")
+	if len(servers) < 1 {
+		return nil, errors.New("invalid server config. at least 1 server needs to be configured")
 	}
-	server := servers[0]
-	tlsConfig = &tls.Config{RootCAs: caPool, InsecureSkipVerify: false, ServerName: server}
-	if TLS {
-		lConn, err = ldapv3.DialTLS("tcp", fmt.Sprintf("%s:%d", server, port), tlsConfig)
-		if err != nil {
-			return nil, fmt.Errorf("Error creating ssl connection: %v", err)
-		}
-	} else if startTLS {
-		lConn, err = ldapv3.Dial("tcp", fmt.Sprintf("%s:%d", server, port))
-		if err != nil {
-			return nil, fmt.Errorf("Error creating connection for startTLS: %v", err)
-		}
-		if err := lConn.StartTLS(tlsConfig); err != nil {
-			return nil, fmt.Errorf("Error upgrading startTLS connection: %v", err)
+	for _, server := range servers {
```

Comment:
will we have UI support for this as well?

Reply:
Yes, the old UI already has the necessary code to accept a comma separated list and split it into a list. For dashboard I had to re-add that feature. See [#5606](https://github.com/rancher/dashboard/pull/5606/files)

Reply:
For reference: 
https://github.com/rancher/ui/blob/543324df8cdb9472eee3950ad3e8b9b45401d749/lib/global-admin/addon/security/authentication/activedirectory/controller.js#L102-L111